silverback/_cli.py

```diff
@@ -142,3 +128,500 @@ def run(cli_ctx, account, runner_class, recorder, max_exceptions, path):
+    type=str,
+    metavar="NAME=VAL",
+    callback=parse_envvars,
+    help="Environment variable key and value to add/update (Multiple allowed)",
+)
+@click.option(
+    "-d",
+    "--del",
+    "deleted_vars",
+    multiple=True,
+    type=str,
+    metavar="NAME",
+    help="Environment variable name to delete (Multiple allowed)",
+)
+@click.argument("name")
+def update_vargroup(
+    client: ClusterClient,
+    name: str,
+    new_name: str,
+    updated_vars: dict[str, str],
+    deleted_vars: tuple[str],
+):
+    """Update a variable GROUP in CLUSTER
+
+    NOTE: Changing the values of variables in GROUP by create a new revision, since variable groups
+    are immutable. New revisions do not automatically update bot configuration."""
+
+    if not (vg := client.variable_groups.get(name)):
+        raise click.UsageError(f"Unknown Variable Group '{name}'")
+
+    if dup := "', '".join(set(updated_vars) & set(deleted_vars)):
+        raise click.UsageError(f"Cannot update and delete vars at the same time: '{dup}'")
+
+    if missing := "', '".join(set(deleted_vars) - set(vg.variables)):
+        raise click.UsageError(f"Cannot delete vars not in group: '{missing}'")
+
+    click.echo(
+        yaml.safe_dump(
+            vg.update(
+                name=new_name,
+                # NOTE: Do not update variables if no updates are provided
+                variables=dict(**updated_vars, **{v: None for v in deleted_vars}) or None,
+            ).model_dump(
+                exclude={"id"}
+            )  # NOTE: Skip machine `.id`
+        )
+    )
+
+
+@vars.command(name="remove")
+@click.argument("name")
+def remove_vargroup(client: ClusterClient, name: str):
+    """
+    Remove a variable GROUP from a CLUSTER
+
+    NOTE: Cannot delete if any bots reference any revision of GROUP
+    """
+    if not (vg := client.variable_groups.get(name)):
+        raise click.UsageError(f"Unknown Variable Group '{name}'")
+
+    vg.remove()  # NOTE: No confirmation because can only delete if no references exist
+    click.secho(f"Variable Group '{vg.name}' removed.", fg="green", bold=True)
+
+
+@cluster.group()
+def bots():
+    """Manage bots in a CLUSTER"""
+
+
+@bots.command(name="new", section="Configuration Commands")
+@click.option("-i", "--image", required=True)
+@click.option("-n", "--network", required=True)
+@click.option("-a", "--account")
+@click.option("-g", "--group", "groups", multiple=True)
+@click.argument("name")
+def new_bot(
+    client: ClusterClient,
+    image: str,
+    network: str,
+    account: str | None,
+    groups: list[str],
+    name: str,
+):
+    """Create a new bot in a CLUSTER with the given configuration"""
+
+    if name in client.bots:
+        raise click.UsageError(f"Cannot use name '{name}' to create bot")
+
+    environment = list()
+    for vg_id in groups:
+        if "/" in vg_id:
+            vg_name, revision = vg_id.split("/")
+            vg = client.variable_groups[vg_name].get_revision(int(revision))
```

Comment:
thought: I'm not much of a fan of users providing revisions. Seems like an unnecessary foot gun. Though I guess we aren't advertising it in the CLI help. :shrug:

Reply:
okay yeah, you are right. it doesn't make sense to support this